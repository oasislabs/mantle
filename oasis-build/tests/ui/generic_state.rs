use oasis_std::Context;

#[derive(oasis_std::Service, Default)]
pub struct State<T>(Option<T>);

impl<T: Default> State<T> {
    pub fn new(ctx: &Context) -> Result<Self, ()> {
        Ok(Default::default())
    }

    fn hmmm() -> Result<(), String> {
        Err(format!("hmm"))
    }
}

fn main() {
    oasis_std::service!(State);
}
