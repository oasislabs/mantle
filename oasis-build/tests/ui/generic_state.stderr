error: RPC definition cannot have generic parameters.
 --> $DIR/generic_state.rs:4:17
  |
4 | pub struct State<T>(Option<T>);
  |                 ^^^

error: aborting due to previous error

